benchmarks/bvh_driver/bvh_driver.cpp

```diff
@@ -328,42 +332,105 @@ int main(int argc, char *argv[])
   std::ignore =
       bpo::command_line_parser(pass_further.argc(), pass_further.argv())
           .options(bpo::options_description(""))
           .run();
 
-  // Google benchmark only supports integer arguments (see
-  // https://github.com/google/benchmark/issues/387), so we map the string to
-  // an enum.
   std::map<std::string, PointCloudType> to_point_cloud_enum;
   to_point_cloud_enum["filled_box"] = PointCloudType::filled_box;
   to_point_cloud_enum["hollow_box"] = PointCloudType::hollow_box;
   to_point_cloud_enum["filled_sphere"] = PointCloudType::filled_sphere;
   to_point_cloud_enum["hollow_sphere"] = PointCloudType::hollow_sphere;
-  int source_point_cloud_type = to_point_cloud_enum.at(source_pt_cloud);
-  int target_point_cloud_type = to_point_cloud_enum.at(target_pt_cloud);
+
+  PointCloudType source_point_cloud_type;
+  PointCloudType target_point_cloud_type;
+
+  auto label_construction = [&](std::string const &tree_name) -> std::string {
+    std::string s = std::string("BM_construction<") + tree_name + ">";
+    for (auto &var : {n_values, (int)source_point_cloud_type})
+      s += "/" + std::to_string(var);
+    return s.c_str();
+  };
+  auto label_knn_search = [&](std::string const &tree_name) -> std::string {
+    std::string s = std::string("BM_knn_search<") + tree_name + ">";
+    for (auto &var :
+         {n_values, n_queries, n_neighbors, sort_predicates_int,
+          (int)source_point_cloud_type, (int)target_point_cloud_type})
+      s += "/" + std::to_string(var);
+    return s.c_str();
+  };
+  auto label_radius_search = [&](std::string const &tree_name) -> std::string {
+    std::string s = std::string("BM_radius_search<") + tree_name + ">";
+    for (auto &var :
+         {n_values, n_queries, n_neighbors, sort_predicates_int, buffer_size,
+          (int)source_point_cloud_type, (int)target_point_cloud_type})
+      s += "/" + std::to_string(var);
+    return s.c_str();
+  };
+
+  if (vm.count("exact-spec") == 0)
+  {
+    exact_specs.resize(1);
+    auto &spec = exact_specs[0];
+    spec = std::to_string(n_values);
+    for (auto &var :
+         {n_queries, n_neighbors, sort_predicates_int, buffer_size,
+          (int)source_point_cloud_type, (int)target_point_cloud_type})
+      spec += "/" + std::to_string(var);
+  }
+
+  for (auto const &spec : exact_specs)
+  {
+    std::istringstream ss(spec);
+    std::string token;
+
+    // clang-format off
+      getline(ss, token, '/');  n_values = std::stoi(token);
+      getline(ss, token, '/');  n_queries = std::stoi(token);
+      getline(ss, token, '/');  n_neighbors = std::stoi(token);
+      getline(ss, token, '/');  sort_predicates_int = std::stoi(token);
+      getline(ss, token, '/');  buffer_size = std::stoi(token);
+      getline(ss, token, '/');  source_point_cloud_type = static_cast<PointCloudType>(std::stoi(token));
+      getline(ss, token, '/');  target_point_cloud_type = static_cast<PointCloudType>(std::stoi(token));
```

Comment:
Not happy about this. The goal here is to tokenize `a/b/c` string and store the results in variables.